functions/api/roles.js

```diff
@@ -2,57 +2,55 @@ var express = require('express')
 var router = express.Router()
 const admin = require('firebase-admin')
 const db = admin.firestore()
-const joi = require('@hapi/joi')
-const validate = require('../common/validator')
-const formatDate = require('../common/dateFormatter')
 const constants = require('../common/constants')
-const { isAdmin } = require('../middleware/auth')
-const audit = require('./audit')
-/*
-1.Creates a new role along with its permissions, status and description
-*/
-router.post('/', isAdmin, async (req, res, next) => {
-    const { error } = validateInput(req.body, constants.CREATE)
-    if (error) {
-        const err = new Error(error.details[0].message)
-        err.statusCode = 400
-        next(err)
-        return
-    }
-    const { name } = req.body
-    let rolesRef = db.collection(constants.ROLES).where('name', '==', name)
-    const snapshot = await rolesRef.get()
-    if (!snapshot.empty) {
-        const err = new Error(`Role ${name} already exists. Please update if needed.`)
-        err.statusCode = 404
-        next(err)
-        return
-    }
-    const data = {
-        ...req.body,
-        createdDate: new Date(),
-        lastUpdatedDate: new Date()
-    }
-    const response = await db.collection(constants.ROLES).add({
-        ...data
-    })
-    // Fire and forget audit log
-    const eventMessage = `User ${req.user.firstName} created new role ${name}`
-    audit.logEvent(eventMessage, req)
+const { isAdminOrSuperAdmin } = require('../middleware/auth')
 
-    res.status(201).send({ id: response.id, ...data })
-})
+// /*
+// 1.Creates a new role along with its permissions, status and description
+// */
+// router.post('/', isAdminOrSuperAdmin, async (req, res, next) => {
+//     const { error } = validateInput(req.body, constants.CREATE)
+//     if (error) {
+//         const err = new Error(error.details[0].message)
+//         err.statusCode = 400
+//         next(err)
+//         return
+//     }
+//     const { name } = req.body
+//     let rolesRef = db.collection(constants.ROLES).where('name', '==', name)
+//     const snapshot = await rolesRef.get()
+//     if (!snapshot.empty) {
+//         const err = new Error(`Role ${name} already exists. Please update if needed.`)
+//         err.statusCode = 404
+//         next(err)
+//         return
+//     }
+//     const data = {
+//         ...req.body,
+//         createdDate: new Date(),
+//         lastUpdatedDate: new Date()
+//     }
+//     const response = await db.collection(constants.ROLES).add({
+//         ...data
+//     })
+//     // Fire and forget audit log
+//     const eventMessage = `User ${req.user.firstName} created new role ${name}`
+//     audit.logEvent(eventMessage, req)
+
+//     res.status(201).send({ id: response.id, ...data })
+// })
```

Comment:
Do we need this commented code block?

Reply:
I decided to have it just in case we go back :) Will remove it if not used towards the end.